docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,387 @@
+{: .caution}
+
+It's possible to configure `kubeadm init` with a configuration file instead of command
+line flags, and some more advanced features may only be available as
+configuration file options.  This file is passed in to the `--config` option.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+api:
+  advertiseAddress: <address|string>
+  bindPort: <int>
+etcd:
+  endpoints:
+  - <endpoint1|string>
+  - <endpoint2|string>
+  caFile: <path|string>
+  certFile: <path|string>
+  keyFile: <path|string>
+  dataDir: <path|string>
+  extraArgs:
+    <argument>: <value|string>
+    <argument>: <value|string>
+  image: <string>
+networking:
+  dnsDomain: <string>
+  serviceSubnet: <cidr>
+  podSubnet: <cidr>
+kubernetesVersion: <string>
+cloudProvider: <string>
+nodeName: <string>
+authorizationModes:
+- <authorizationMode1|string>
+- <authorizationMode2|string>
+token: <string>
+tokenTTL: <time duration>
+selfHosted: <bool>
+apiServerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+controllerManagerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+schedulerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+apiServerCertSANs:
+- <name1|string>
+- <name2|string>
+certificatesDir: <string>
+imageRepository: <string>
+unifiedControlPlaneImage: <string>
+featureGates:
+  <feature>: <bool>
+  <feature>: <bool>
+```
+
+### Using custom images {#custom-images}
+
+By default, kubeadm will pull images from `gcr.io/google_containers`, unless
+requested kubernetes version is a ci version; in this case
+`gcr.io/kubernetes-ci-image` will be used.
+
+This behaviour can be overridden by [using kubeadm with a configuration file](#config-file).
+Allowed customization are:
+
+* provide an alternative `imageRepository` to be used instead of
+  `gcr.io/google_containers`
+* provide an `unifiedControlPlaneImage` to be used instead of different images for each control plane component
+  for control plane components
+* provide a specific `etcd.image` to be used instead of the image available at`gcr.io/google_containers`
+
+
+### Using custom certificates {#custom-certificates}
+
+By default kubeadm will generate all the certificates needed for a cluster to run.
+You can override this behaviour by providing your own certificates.
+
+To do so, you must place them in whatever directory is specified by the
+`--cert-dir` flag or `CertificatesDir` configuration file key. By default this
+is `/etc/kubernetes/pki`.
+
+If a given certificate and private key pair both exist, kubeadm will skip the
+generation step and those files will be validated and used for the prescribed
+use-case.
+
+This means you can, for example, copy an existing CA into `/etc/kubernetes/pki/ca.crt`
+and `/etc/kubernetes/pki/ca.key` and kubeadm will use this CA for signing the rest 
+of the certs.
+
+Only for the CA certificate, it is possible to provide the `ca.crt` file but not the 
```

Comment:
It is also possible to provide just `ca.crt` and not the `ca.key` file (this is only available for the root CA file, not other cert pairs). If all other